src/katgpucbf/kernels/curand_helpers.mako

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2024, National Research Foundation (SARAO)
+ *
+ * Licensed under the BSD 3-Clause License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ *   https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+<%include file="/port.mako"/>
+
+extern "C++"  // PyCUDA wraps the whole file in extern "C"
+{
+#include <curand_kernel.h>
+#include <type_traits>
+#include <utility>
+}
+
+/* Holds the same random state as curandStateXORWOW_t, excluding the state for
+ * the Box-Muller transform. It uses an array of uint2 rather than uint so
+ * that global memory instructions use 64-bit rather than 32-bit access.
```

Comment:
Aha, this is where the magic sauce is used to reduce memory traffic - moving to `uint2`.

Reply:
No actually - using uint2 is to make the memory traffic that exists require fewer instructions; I expect that to make a very difference if any. The reduction comes from using a smaller structure. Here's curandStateXORWOW_t:
```c++
struct curandStateXORWOW {
    unsigned int d, v[5];
    int boxmuller_flag;
    int boxmuller_flag_double;
    float boxmuller_extra;
    double boxmuller_extra_double;
};
```
We're jettisoning all the boxmuller fields since they're not needed for uniform random distributions.